enhancements/sig-architecture/70-spread-policy/README.md

```diff
@@ -0,0 +1,288 @@
+	// +required
+	// +kubebuilder:validation:Required
+	TopologyKeyType TopologyKeyType `json:"topologyKeyType"`
+
+	// MaxSkew describes the degree to which the workload may be unevenly distributed.
+	// If not omitted, the scheduler would not schedule if MaxSkew cannot be satisfied.
+	// The minimum possible value is 1.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	MaxSkew *int32 `json:"maxSkew,omitempty"`
+
+	// Weight indicates the weight of the SpreadPolicyTerm. It is required only when multiple
+	// terms are used. To spread the workload, the scheduler takes the even constraints with
+	// higher weight into prior consideration.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=1
+	Weight int32 `json:"weight,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=Label;Claim
+type TopologyKeyType string
+
+const (
+	// Valid TopologyKeyType value is Claim, Label.
+	TopologyKeyTypeClaim TopologyKeyType = "Claim"
+	TopologyKeyTypeLabel TopologyKeyType = "Label"
+)
+```
+
+### API Example
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: demo-placement
+spec:
+  # numberOfClusters is required if spreadConstraints contains
+  # a constraint of Even type.
+  numberOfClusters: 4
+  # Prediates work before the spread constraints to filter
+  # out some undesired topologies.
+  predicates:
+  - # ...
+  spreadPolicy:
+    # spreadConstraints contain two terms, which work
+    # independently and jointly.
+    spreadConstraints:
+    # When multiple terms are used, the user should specify their weights.
+    # The scheduler will consider to first do the spread according to the
+    # terms with higher weight.
+    # In this example, the scheduler will first spread the workload
+    # by providers.
+    # Within each provider, the scheduler then spread the workload by
+    # regions. 
+    - topologyKey: provider
+      topologyKeyType: Label
+      maxSkew: 2
+      weight: 20
+      # maxSkew is omitted as this term is a soft constraint
+    - topologyKey: region
+      topologyKeyType: Label
+      weight: 10
+  # The spread policy has a scoreCoordinate in prioritizerPolicy.
+  # In Additive mode, the weight of SpreadPolicy is 0 by default.
+  # In Exact mode, the weight of SpreadPolicy should be set by users.
+  prioritizerPolicy:
+    mode: Exact
+    configurations:
+      - scoreCoordinate:
+          builtIn: ResourceAllocatableMemory
+      - scoreCoordinate:
+          builtIn: SpreadPolicy
+        weight: 2
+```
+
+### Implementation
+The current scheduling framework contains two types of plugins, i.e., `Filter`s and `Prioritizer`s. 
+`Filter`s are invoked *before* `Prioritizer`s to handle hard constrains, filtering the clusters. 
+Then, the `Prioritizer`s score the clusters.
+To implement `maxSkew`, we need to first score the clusters (i.e., do the spread), then ensure the `maxSkew` is not exceeded by some filter-like operations for hard constraints.
```

Comment:
What filter are you thinking of?  The total number of filtered results is not applied until after prioritization.

Reply:
Examples for filters here is Predicate and TaintToleration.